jerry-main/cli.h

```diff
@@ -0,0 +1,151 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CLI_H
+#define CLI_H
+
+#include <string.h>
+
+/*
+ * Types for CLI
+ */
+
+/**
+ * Command line option definition
+ */
+typedef struct
+{
+  int id; /**< unique ID of the option (CLI_OPT_END, CLI_OPT_POSITIONAL, or anything >= 0) */
+  const char *opt; /**< short option variant (in the form of "-x") */
+  const char *longopt; /**< long option variant (in the form of "--xxx") */
+  int argc; /**< number of arguments of the option */
+  const char *meta; /**< name(s) of the argument(s) of the option, for display only */
+  int quant; /**< quantifier of the option (CLI_QUANT_{Q,A,P,1} for ?,*,+,1), for display only */
+  const char *help; /**< descriptive help message of the option */
+} cli_opt_t;
+
+/**
+ * Quantifiers of command line options
+ */
+typedef enum
+{
+  CLI_QUANT_Q, /**< ? (Question mark: optional, zero or one) */
+  CLI_QUANT_A, /**< * (Asterisk, star: zero or one or more) */
+  CLI_QUANT_P, /**< + (Plus sign: one or more) */
+  CLI_QUANT_1  /**< 1 (One: one) */
```

Comment:
No need for that, as there is no use case other than these, which could be supported easily. Note that the quantifiers are for display only, not enforced in any other way.
* `?` gets displayed as `[-x X]` or `[X]`
* `*` gets displayed as `[-x X]...` or `[X]...`
* `+` gets displayed as `-x X...` or `X...`
* `1` gets displayed as `-x X` or `X`

There is no easy or conventional notation for more general quantifiers (like min 3, max infinity) for command line arguments, and I'm not sure we'd need them. (This is not a general purpose CLI library, it's for the needs of jerry-main only.)